pkg/cloudprovider/providers/gce/gce_disks.go

```diff
@@ -511,6 +511,11 @@ func newDiskMetricContextRegional(request, region string) *metricContext {
 }
 
 func (gce *GCECloud) GetLabelsForVolume(ctx context.Context, pv *v1.PersistentVolume) (map[string]string, error) {
+	// Ignore any volumes which are not GCEPersistentDisk
+	if pv.Spec.GCEPersistentDisk == nil {
+		return nil, nil
```

Comment:
I'm a little worried that the caller (pkg/controller/cloud/pvlcontroller.go:addLabelsAndAffinityToVolume) will wipe out the volume labels if they had been set.

Reply:
This behavior was there for a quite a bit - azure had it since 3 months ago:
https://github.com/kubernetes/kubernetes/blame/b5c25728ec4e740a5f880b3d39ef91993276cc9d/pkg/cloudprovider/providers/azure/azure_managedDiskController.go#L238

This problem also applies for the next check as well:

https://github.com/kubernetes/kubernetes/blob/6f897af2daffd8cd3539f1ef279b9f9bc280d98f/pkg/cloudprovider/providers/gce/gce_disks.go#L515

If there the volume is being provisioned, then the labels would be removed as well.

@cheftako Maybe it's better to change the behavior of the controller and add check if returned labels are nil.

Reply:
@cheftako I've added a PR #70533 for this exact behavior